pkg/comp-functions/functions/vshnnextcloud/collabora.go

```diff
@@ -0,0 +1,732 @@
+}
+func AddCollaboraService(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+
+	service := &corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels: map[string]string{
+				"app": comp.GetName() + "-collabora-code",
+			},
+		},
+		Spec: corev1.ServiceSpec{
+			Selector: map[string]string{"app": comp.GetName() + "-collabora-code"},
+			Ports: []corev1.ServicePort{
+				{
+					Port:       9980,
+					TargetPort: intstr.FromInt(9980),
+				},
+			},
+		},
+	}
+
+	return svc.SetDesiredKubeObject(service, comp.GetName()+"-collabora-code-service", runtime.KubeOptionAddLabels(labelMap))
+
+}
+func AddCollaboraIngress(comp *vshnv1.VSHNNextcloud, svc *runtime.ServiceRuntime) error {
+	annotations := map[string]string{}
+	if svc.Config.Data["collabora_ingress_annotations"] != "" {
+
+		err := yaml.Unmarshal([]byte(svc.Config.Data["collabora_ingress_annotations"]), annotations)
+		if err != nil {
+			svc.Log.Error(err, "cannot unmarshal ingress annotations from input")
+			svc.AddResult(runtime.NewWarningResult(fmt.Sprintf("cannot unmarshal ingress annotations from input: %s", err)))
+		}
+	}
+	annotations["route.openshift.io/termination"] = "reencrypt"
+	annotations["route.openshift.io/destination-ca-certificate-secret"] = comp.GetName() + "-collabora-code-ingress-certificate"
+
+	ingress := &networkingv1.Ingress{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      comp.GetName() + "-collabora-code",
+			Namespace: comp.GetInstanceNamespace(),
+			Labels: map[string]string{
+				"app": comp.GetName() + "-collabora-code",
+			},
+			Annotations: annotations,
+		},
+		Spec: networkingv1.IngressSpec{
+			Rules: []networkingv1.IngressRule{
+				{
+					Host: comp.Spec.Parameters.Service.Collabora.FQDN,
+					IngressRuleValue: networkingv1.IngressRuleValue{
+						HTTP: &networkingv1.HTTPIngressRuleValue{
+							Paths: []networkingv1.HTTPIngressPath{
+								{
+									Path:     "/",
+									PathType: ptr.To(networkingv1.PathTypePrefix),
+									Backend: networkingv1.IngressBackend{
+										Service: &networkingv1.IngressServiceBackend{
+											Name: comp.GetName() + "-collabora-code",
+											Port: networkingv1.ServiceBackendPort{
+												Number: 9980,
+											},
+										},
+									},
+								},
+							},
+						},
+					},
+				},
+			},
+			TLS: []networkingv1.IngressTLS{
+				{
+					Hosts:      []string{comp.Spec.Parameters.Service.Collabora.FQDN},
+					SecretName: comp.GetName() + "-collabora-code-tls",
+				},
+			},
+		},
+	}
+
+	if svc.Config.Data["ingress_annotations"] != "" {
+		annotations := map[string]string{}
+
+		err := yaml.Unmarshal([]byte(svc.Config.Data["ingress_annotations"]), annotations)
+		if err != nil {
+			return fmt.Errorf("cannot unmarshal ingress annotations from input: %w", err)
+		}
+
+		if val, ok := annotations["cert-manager.io/cluster-issuer"]; ok {
```

Comment:
Why are you only setting a single annotation here? `svc.Config.Data["ingress_annotations"]` could contain multiple annotations for the ingress.

Reply:
I guess we just need to assigned the whole map, no need for if clauses

Reply:
refactored it, the issue came from different approaches how to handle those annotations, should be good now, leaving as unresolved until I test that in LAB